src/actions/notifications.rs

```diff
@@ -73,7 +74,7 @@ impl BlockingNotificationAction for DidOpenTextDocument {
 }
 
 impl BlockingNotificationAction for DidChangeTextDocument {
-    fn handle<O: Output>(params: Self::Params, ctx: &mut InitActionContext, out: O) -> Result<(), ()> {
+    fn handle<O: Output>(params: Self::Params, jobs: &mut Jobs, ctx: &mut InitActionContext, out: O) -> Result<(), ()> {
```

Comment:
could `jobs` be part of the context here, rather than adding a variable?

Reply:
Moved jobs to the Context. Initially I've avoided that because `Context` is clone, and so `Arc<Mutex>` is required. And having explicit `&mut Jobs` argument and `-> ConcurrentJob` return types gives better idea about when concurrency happens. However, that is indeed a lot of extra typing compared to just stuffing jobs into the ctx.